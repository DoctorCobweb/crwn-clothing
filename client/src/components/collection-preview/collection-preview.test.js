import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './collection-preview.component';

describe('CollectionPreview componnet testing', ()=>{
  let wrapper;
  let mockHistory;
  const mockTitle = 'blah';
  const mockItems = [{id:0}, {id:1}, {id:2}, {id:3}, {id:4}, {id:5}, {id:6}];
  const mockMatch = { path: 'mockPath' };
  const mockRouteName = 'mockRouteName';

  beforeEach(()=>{
    mockHistory = {
      push: jest.fn(),
    }
    const mockProps = {
      title: mockTitle,
      items: mockItems,
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
    };
    wrapper = shallow(<CollectionPreview {...mockProps }/>);
  });

  it('component should snapshot should match', ()=>{
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('when clicking title container, correctly navigate to collection route', () => {
    wrapper.find('TitleContainer').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.path}/${mockRouteName}`);
  });

  it('TitleContainer text should be set and uppercased', () => {
    const titleText = wrapper.find('TitleContainer').text();
    expect(titleText).toBe(mockTitle.toUpperCase());
  });

  it('PreviewContainer should limit to 4 items to preview', () => {
    expect(wrapper.find('PreviewContainer').children().length).toBe(4);
  });

})