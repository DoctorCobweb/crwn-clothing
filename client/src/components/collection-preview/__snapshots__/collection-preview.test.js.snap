// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview componnet testing component should snapshot should match 1`] = `
"<CollectionPreviewContainer>
  <TitleContainer onClick={[Function: onClick]}>
    BLAH
  </TitleContainer>
  <PreviewContainer>
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
  </PreviewContainer>
</CollectionPreviewContainer>"
`;
